.travis.yml

```diff
@@ -21,23 +20,21 @@ env:
     - FLAVOR="cpp-comm" BOOST="1.63.0"
     - FLAVOR="cpp-grpc" BOOST="1.64.0"
     - FLAVOR="cpp-grpc" BOOST="1.63.0"
-    - FLAVOR="hs" GHC_VERSION="7.8.4"
-    - FLAVOR="hs" GHC_VERSION="7.6.3"
+    - FLAVOR="hs"
 addons:
   apt:
     sources:
         - ubuntu-toolchain-r-test
         - hvr-ghc
     packages:
         - g++-4.8
-        - cabal-install-1.22
-        - ghc-7.8.4
-        - ghc-7.6.3
-        - ghc
         - ccache
         - wget
 
 before_install:
+    - mkdir -p ~/.local/bin
+    - export PATH=$HOME/.local/bin:$PATH
+    - travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
```

Comment:
Better leave it as is in case of intermittent network issues. https://blog.travis-ci.com/2013-05-20-network-timeouts-build-retries/